executor/logging.go

```diff
@@ -4,21 +4,131 @@ import (
 	"bufio"
 	"io"
 	"log"
+	"time"
 )
 
+var stdLogBufferSize = 64 * 1024
+
 // bindLoggingPipe spawns a goroutine for passing through logging of the given output pipe.
+//
+// This implementation is adapted from https://github.com/hashicorp/go-plugin/pull/98
 func bindLoggingPipe(name string, pipe io.Reader, output io.Writer) {
 	log.Printf("Started logging %s from function.", name)
 
-	scanner := bufio.NewScanner(pipe)
-	logger := log.New(output, log.Prefix(), log.Flags())
+	scanner := bufio.NewReaderSize(pipe, stdLogBufferSize)
 
 	go func() {
-		for scanner.Scan() {
-			logger.Printf("%s: %s", name, scanner.Text())
-		}
-		if err := scanner.Err(); err != nil {
-			log.Printf("Error scanning %s: %s", name, err.Error())
+		logFlags := log.Flags()
+		continuation := false
+
+		for {
+			line, isPrefix, err := scanner.ReadLine()
```

Comment:
I would expect to see the partial lines added to a local string and then written out with the same approach as before such as: `logger.Printf("%s: %s", name, scanner.Text())`. This would mean not needing to duplicate the stdlib Go code and to maintain it here and in the classic watchdog.

Reply:
```
	var b bytes.Buffer
	b.WriteString()
```

The 64KB is an outlier for a log message, and I'm unaware of any need to support multiple-mb log lines.

Reply:
@alexellis i have pushed a new version that does not include that code from the stdlib

I would still prefer to avoid creating another buffered copy of the log if we can avoid it